lib/core/src/Cardano/Wallet/Api/Server.hs

```diff
@@ -2477,6 +2481,20 @@ submitTransaction ctx apiw@(ApiT wid) apitx@(ApiSerialisedTransaction (ApiT seal
             all isInpForeign generalInps &&
             all isWdrlForeign generalWdrls
 
+    sameInput inp1 inp2 = case (inp1, inp2) of
+        (WalletInput (ApiWalletInput _ _ _ derPath1 _ _), WalletInput (ApiWalletInput _ _ _ derPath2 _ _) ) ->
+               derPath1 == derPath2
+        (ExternalInput txin1, ExternalInput txin2) ->
+               txin1 == txin2
+        _ -> False
+
+    certWitNeeded = \case
+        WalletDelegationCertificate (JoinPool _ _) -> True
+        WalletDelegationCertificate (QuitPool _) -> True
+        DelegationCertificate (JoinPoolExternal _ _) -> True
+        DelegationCertificate (QuitPoolExternal _) -> True
+        _ -> False
```

Comment:
registration of stake credential does not need witness. Also I omitted witnesses from pool ops, mirs, etc.